client/python/cryoet_data_portal/src/cryoet_data_portal/_gql_base.py

```diff
@@ -211,10 +211,19 @@ def __init__(self, client: Client, **kwargs):
             value = getattr(self, k).convert(kwargs.get(strcase.to_lower_camel(k)))
             setattr(self, k, value)
 
+    def _serialize(self, value):
+        if isinstance(value, (datetime, date)):
+            return value.isoformat()
+        return value
+
     def to_dict(self) -> Dict[str, Any]:
         """Return a dictionary representation of this object's attributes"""
         return {k: getattr(self, k) for k in self._get_scalar_fields()}
 
+    def to_json_dict(self) -> Dict[str, Any]:
+        """Return a JSON encoder friendly representation of this object's attributes"""
+        return {key: self._serialize(val) for key, val in self.to_dict().items()}
```

Comment:
Python's `json` module already supplies a way to convert python values to json-compatible values ([docs here](https://docs.python.org/3/library/json.html#json.JSONEncoder.default) and [example implementation here](https://stackoverflow.com/a/35870294) ). Do we need to implement this functionality in our client, or can we leave it to users to work with python's standard library?

Reply:
cc @uermel